test_run.sh

```diff
@@ -68,15 +68,16 @@ test_expect_failure_or_warning_on_missing_direct_deps_with_expected_message() {
 
   expected_message=$1
   test_target=$2
-  operator=${3:-"eq"}
+  strict_deps_mode=$3
+  operator=${4:-"eq"}
```

Comment:
WDYT about adding an alias to eq, ne
something like expecting_failure, expecting_success